radicle-node/src/service.rs

```diff
@@ -1083,39 +1116,48 @@ where
                     // TODO: We should fetch here if we're already connected, case this seed has
                     // refs we don't have.
                 }
-                updated = true;
+                updated.push(*proj_id);
             }
         }
         for id in self.routing.get_resources(&from)?.into_iter() {
             if !included.contains(&id) {
                 if self.routing.remove(&id, &from)? {
-                    updated = true;
+                    updated.push(id);
                 }
             }
         }
         Ok(updated)
     }
 
     /// Announce local refs for given id.
-    fn announce_refs(&mut self, id: Id) -> Result<(), storage::Error> {
-        type Refs = BoundedVec<Id, REF_LIMIT>;
-
-        let node = self.node_id();
-        let repo = self.storage.repository(id)?;
-        let remote = repo.remote(&node)?;
+    fn announce_refs(&mut self, rid: Id, namespaces: Namespaces) -> Result<(), storage::Error> {
+        let repo = self.storage.repository(rid)?;
         let peers = self.sessions.negotiated().map(|(_, p)| p);
-        let timestamp = self.clock.as_secs();
-
-        if remote.refs.len() > Refs::max() {
-            error!(
-                target: "service",
-                "refs announcement limit ({}) exceeded, other nodes will see only some of your project references",
-                Refs::max(),
-            );
+        let timestamp = self.clock.as_millis();
+        let mut refs = BoundedVec::<_, REF_REMOTE_LIMIT>::new();
+
+        match namespaces {
+            Namespaces::All => {
+                for (remote_id, remote) in repo.remotes()?.into_iter() {
+                    if refs.push((remote_id, remote.refs.unverified())).is_err() {
+                        warn!(
+                            target: "service",
+                            "refs announcement limit ({}) exceeded, peers will see only some of your repository references",
+                            REF_REMOTE_LIMIT,
+                        );
+                        break;
+                    }
+                }
+            }
+            Namespaces::One(pk) => refs
+                .push((pk, repo.remote(&pk)?.refs.unverified()))
+                // SAFETY: `REF_LIMIT` is greater than 1, thus the bounded vec can hold at least
```

Comment:
You changed this name, right?
```suggestion
                // SAFETY: `REF_REMOTE_LIMIT` is greater than 1, thus the bounded vec can hold at least
```

Reply:
Indeed, let me fix!